core/app/models/spree/promotion_code/code_builder.rb

```diff
@@ -26,7 +26,7 @@ def build_promotion_codes
   private
 
   def codes
-    if num_codes > 1
+    if (num_codes + promotion.codes.size) > 1
```

Comment:
Prior to this change, if you tell the CodeBuilder to create one code, it _insists_ on creating that code _without_ a suffix -- the one code created is just the 'base_code' by itself. 

Even ignoring actual admin UI, I had tried to use the CodeBuilder itself to create _one_ _additional_ code, with suffix -- I could not do so, it won't do it. 

The CodeBuilder was I guess written with the assumption that it would only be used on new promotion creation. 

So this change keeps the intent and backwards compatible API of the CodeBuilder -- which seems to be "Decide whether to use a suffix based on how many codes there are" -- but makes it work for both about-to-be-created new promotions, and existing promotions. For existing promtions "how many codes there are" needs to take into account not just the codes you are about to add, but the codes that existed already as well. 

Again, this is not the API I would have done, but it's what's there, and I'm trying to not rewrite the whole thing, and maintain backwards compatibility. 

I am happy to ammend the commit message to be whatever you want if you have a suggestion.